parsec/core/local_storage.py

```diff
@@ -117,6 +117,12 @@ def set_manifest(self, entry_id: EntryID, manifest: LocalManifest) -> None:
         self.persistent_storage.set_manifest(entry_id, raw)
         self.local_manifest_cache[entry_id] = manifest
 
+    def set_manifest_cache_only(self, entry_id: EntryID, manifest: LocalManifest) -> None:
+        assert isinstance(entry_id, EntryID)
+        assert manifest.author == self.device_id
+        self._check_lock_status(entry_id)
+        self.local_manifest_cache[entry_id] = manifest
```

Comment:
This method could also store the `entry_id` in a set of cache-only ids. The `set_manifest` method would also discard the corresponding id from this set. Then a `set_manifest` call have to be issued for each id in the set when `__exit__`ing the local storage.

Reply:
I've modified the code to keep a set of ids and flushing them in LocalStorage's `__exit__`

However I'm still not sure how to handle `LocalStorage.clear_manifest` & `LocalStorage.clear_memory_cache`.
They are only used during test so I'm wondering if you shouldn't have a `TestLocalStorage` child class implementing those methods instead
I've considered their desired behavior is to get rid of the  data, so we lose it if it was only in cache.